api/napi/src/types/image_data.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use i_slint_core::{
+    graphics::{Image, SharedImageBuffer, SharedPixelBuffer},
+    ImageInner,
+};
+use napi::bindgen_prelude::{Buffer, External};
+
+#[napi(js_name = ImageData)]
+pub struct JsImageData {
+    inner: Image,
+}
+
+impl From<Image> for JsImageData {
+    fn from(image: Image) -> Self {
+        Self { inner: image }
+    }
+}
+
+#[napi]
+impl JsImageData {
+    #[napi(constructor)]
+    pub fn new(width: u32, height: u32) -> Self {
+        Self { inner: Image::from_rgba8(SharedPixelBuffer::new(width, height)) }
+    }
+
+    // FIXME: constructor with buffer causes trouble
+    #[napi(constructor)]
+    pub fn from_data_array(data_array: Buffer, width: u32) -> Self {
+        Self {
+            inner: Image::from_rgba8(SharedPixelBuffer::clone_from_slice(
+                data_array.as_ref(),
+                width,
+                width / data_array.len() as u32,
+            )),
+        }
+    }
+
+    #[napi(getter)]
+    pub fn width(&self) -> u32 {
+        self.inner.size().width
+    }
+
+    #[napi(getter)]
+    pub fn height(&self) -> u32 {
+        self.inner.size().height
+    }
+
+    #[napi(getter)]
+    pub fn data(&self) -> Buffer {
+        let image_inner: &ImageInner = (&self.inner).into();
+        if let Some(buffer) = image_inner.render_to_buffer(None) {
+            match buffer {
+                SharedImageBuffer::RGB8(buffer) => return Buffer::from(buffer.as_bytes()),
+                SharedImageBuffer::RGBA8(buffer) => return Buffer::from(buffer.as_bytes()),
+                SharedImageBuffer::RGBA8Premultiplied(buffer) => {
```

Comment:
I think for RGB8 and RGBA8Premultiplied the implementation is wrong. According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/ImageData/data) the `data` property is "stored as a one-dimensional array in the RGBA order, with integer values between 0 and 255 (inclusive)".